ui/app/AppLayouts/Chat/controls/community/CommunityPermissionsRow.qml

```diff
@@ -0,0 +1,315 @@
+                shortModel.append(permission1)
+
+                // More than 2 permissions but 1st and 2nd have only 1:1 or 1:2 tokens
+                if((holdingsCount1 === 1 && holdingsCount2 === 1) ||
+                   (holdingsCount1 === 2 && holdingsCount2 === 1) ||
+                   (holdingsCount1 === 1 && holdingsCount2 === 2)) {
+                    shortModel.append(permission2)
+                }
+            }
+        }
+    }
+
+    implicitHeight: 24
+    spacing: 4
+    padding: 1
+
+    background: Rectangle {
+        color: root.backgroundColor
+        radius: root.backgroundRadius
+    }
+
+    contentItem: RowLayout {
+        id: container
+
+        anchors.centerIn: parent
+        anchors.margins: root.padding
+        spacing: root.spacing
+
+        StatusIcon {
+            Layout.preferredHeight: container.height - 6
+            Layout.preferredWidth: Layout.preferredHeight
+            Layout.leftMargin: 4
+
+            icon: root.requirementsMet ? "tiny/unlocked" : "tiny/locked"
+            color: Theme.palette.baseColor1
+        }
+
+        Repeater {
+            id: repeater
+
+            model: shortModel
+
+            RowLayout {
+                spacing: root.spacing
+
+                SinglePermissionRow {
+                    model: holdingsListModel
+                }
+
+                StatusBaseText {
+                    visible: index !== (repeater.count - 1) || d.dotsVisible
+                    text: qsTr("or")
+                    font.pixelSize: root.pixelSize
+                    color: Theme.palette.baseColor1
+                }
+            }
+        }
+
+        StatusRoundedComponent {
+            Layout.preferredHeight: container.height
+            Layout.preferredWidth: Layout.preferredHeight
+
+            visible: d.dotsVisible
+            color: Theme.palette.baseColor3
+            border.color: root.backgroundColor
+            border.width: root.overlappingBorder
+
+            StatusIcon {
+                anchors.centerIn: parent
+                visible: d.dotsVisible
+                icon: "dots-icon"
+                height: root.dotsIconSize
+                width: height
+            }
+        }
+    }
+
+    onModelChanged: d.buildShortModel(root.model)
+    Component.onCompleted: d.buildShortModel(root.model)
+
+    ListModel{ id: shortModel }
+
+    component SinglePermissionRow :  RowLayout {
+        id: singlePermissionItem
+
+        readonly property int maxVisualTokens: 3
+
+        property var model
+        property string plusElementText: ""
+        property bool plusElementVisible: false
+
+        function buildTokensRowModel(model) {
+            shortTokensRowModel.clear()
+            if(!model)
+                return
+
+            var modelCount = model.rowCount()
+
+            // Need of shorter model
+            if(modelCount > maxVisualTokens) {
+                for(var i = 0; i < singlePermissionItem.maxVisualTokens - 1; i++)
```

Comment:
> I understand the `maxVisualTokens` is 2, not 3. It this right?

No, it is 3 and if there are more tokens, then 2 are displayed with the additional `+` button.
![Screenshot 2023-04-27 at 12 54 11](https://user-images.githubusercontent.com/97019400/234842282-0ba3c299-0018-4635-bc27-47b03c543e1f.png)

Reply:
> ```
>      for(var i = 0; i < Math.min(singlePermissionItem.maxVisualTokens, modelCount); i++) {
>              shortTokensRowModel.append(ModelUtils.get(model, i))
>      }
>      
>      plusElementVisible = modelCount > maxVisualTokens
>      if(plusElementVisible)
>           plusElementText = qsTr("+%1").arg(modelCount - maxVisualTokens)
> ```

I'll added `getVisualTokensCount()` to calculate the visual tokens depending on the case, since this proposal doesn't fit exactly with requirements but can be used anyway to improve the code!! Thanks